downstairs/src/lib.rs

```diff
@@ -1039,20 +1039,35 @@ where
                 let mut abort_needed = false;
                 if let Some(m) = m {
                     abort_needed = check_message_for_abort(&m);
-                    ads.lock()
-                        .await
-                        .complete_work_stat(upstairs_connection, &m, job_id)
-                        .await?;
-                    // Notify the upstairs before completing work
-                    let mut fw = fw.lock().await;
-                    fw.send(&m).await?;
-                    drop(fw);
-
-                    ads.lock()
-                        .await
-                        .complete_work(upstairs_connection, job_id, m)
-                        .await?;
-                    cdt::work__done!(|| job_id);
+
+                    if m.err() {
```

Comment:
If we reset to new here, it won't retry right away, but put the job back on
the list as "new", much like what a lossy would do sometimes, right?
And it will retry it, after having walked the entire list in `new_work`?

If we have just one job, and that one job fails, I don't think we will go back through
the loop, right?  Because I believe we need to see `job_channel_rx.recv().await.is_some()`
to take another lap through the work queue.

Reply:
I changed the logic to loop instead of setting the state back to New (0970a29b90ea7723d8eec169bc32134367bac17c) This should retry jobs until they pass without needing another `job_channel_rx.recv().await.is_some()`